theano/gof/opt.py

```diff
@@ -1312,7 +1312,7 @@ def apply_mult_opts(node, fgraph, multiple_opts=False, prev_repl=None):
                     # Ensuring not the input of graph
                     assert repl[0].owner
                     new_node = repl[0].owner
-                    apply_mult_opts(new_node, fgraph, True, repl)
+                    repl = apply_mult_opts(new_node, fgraph, True, repl)
```

Comment:
That won't work. It will cause all optimization to be skiped. repl will be None. I think it should be:

new_repl = apply...
if new_repl:
    repl = new_repl

Test it. Make a break point and verify manually that this is doing what we want.